### PR TITLE
Make RequestCreator non-final

### PR DESCRIPTION
**Pull request by jrodbx**

### CODE REVIEW

File: picasso/src/test/java/com/squareup/picasso3/TestUtils.java [hunk 1]
```diff
@@ -212,10 +211,14 @@ static Callback mockCallback() {
...
-  static DeferredRequestCreator mockDeferredRequestCreator(ImageView target) {
+  static DeferredRequestCreator mockDeferredRequestCreator(RequestCreator creator, ImageView target) {
     ViewTreeObserver observer = mock(ViewTreeObserver.class);
     when(target.getViewTreeObserver()).thenReturn(observer);
-    return new DeferredRequestCreator(mock(RequestCreator.class, withSettings().useConstructor()), target, null);
+    return new DeferredRequestCreator(creator, target, null);
+  }
+
+  static RequestCreator mockRequestCreator() {
+    return new RequestCreator(mock(Picasso.class), null, 0);
```
Comment (chris-horner) [inline picasso/src/test/java/com/squareup/picasso3/TestUtils.java:221]: Sneaky 😛

**Review by chris-horner — APPROVED** [review]
Massive nit: Commit message should be something like `Make RequestCreator final`

**Comment by jrodbx** [comment]
> Massive nit: Commit message should be something like `Make RequestCreator final`

🤦‍♂️  fixed

File: picasso/src/test/java/com/squareup/picasso3/TestUtils.java (hunk 1 above)
Comment (jrodbx) [reply picasso/src/test/java/com/squareup/picasso3/TestUtils.java:221]: 😉 start small; incremental progress

**Pull request merged**